### PR TITLE
fix xarray provider date handling

### PR DESCRIPTION
**Pull request by tomkralidis**
# Overview
Safeguards fix against newer cftime installations.  Also moves flake8 testing later on in CI.
# Related Issue / Discussion
#993 
# Additional Information

# Contributions and Licensing

(as per https://github.com/geopython/pygeoapi/blob/master/CONTRIBUTING.md#contributions-and-licensing)

- [x] I'd like to contribute [feature X|bugfix Y|docs|something else] to pygeoapi. I confirm that my contributions to pygeoapi will be compatible with the pygeoapi license guidelines at the time of contribution.
- [x] I have already previously agreed to the pygeoapi Contributions and Licensing Guidelines

### CODE REVIEW

File: pygeoapi/provider/xarray_.py [hunk 1]
```diff
@@ -554,11 +555,16 @@ def _to_datetime_string(datetime_obj):
     :returns: str representation of datetime
     """
 
+    LOGGER.warning(f"datetime value {datetime_obj}")
+    LOGGER.warning(f"datetime type {type(datetime_obj)}")
```
Comment (justb4) [inline pygeoapi/provider/xarray_.py:559]: Do these LOGGER warnings need to be present? Maybe remainders from debugging? Also double quoted str.

**Review by justb4 — COMMENTED** [review]
Would be great if this fixes the 'get_oas' error in the unit test on GH Workflow. If that is the case, happy to approve/merge.

File: pygeoapi/provider/xarray_.py (hunk 1 above)
Comment (tomkralidis) [reply pygeoapi/provider/xarray_.py:559]: No, they are for debugging purposes ATM. Will remove once tests pass in this PR.

**Comment by tomkralidis** [comment]
@justb4 @webb-ben CI should now be fixed.  Feel free to review/approve/merge.

**Review by justb4 — APPROVED** [review]
Ok, looks good, let's see if this is the last test to be fixed. This should fix the `get_oas()` error as in #993 .

**Pull request merged**